activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -516,6 +521,20 @@ def default_index_type?(index) # :nodoc:
 
       private
 
+        def with_connection_verify(exception_class)
+          tries = 3
```

Comment:
Why 3 tries?  If we are just trying to reconnect to handle lost connections, then it should only need a single retry to recover from that type of error.